javaharness/java/arcs/api/ParticleExecutionContext.java

```diff
@@ -1,11 +1,54 @@
 package arcs.api;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
+import javax.inject.Inject;
 
-public interface ParticleExecutionContext {
-  void initializeParticle(
+public class ParticleExecutionContext {
+  private PortableJsonParser jsonParser;
+  private ArrayList<Particle> particles;
+  private HandleFactory handleFactory;
+
+  @Inject
+  public ParticleExecutionContext(
```

Comment:
optional - we can get rid of this class altogether and just move the `initializeParticle` method to be a private method in PecInnerPort.